pkg/content/opts.go

```diff
@@ -47,10 +47,13 @@ func WithOutputHash(hash digest.Digest) WriterOpt {
 }
 
 // WithBlocksize set the blocksize used by the processor of data.
-// The default is DefaultBlocksize, which is the same as that used by io.Copy
+// The default is DefaultBlocksize, which is the same as that used by io.Copy.
+// Includes a safety check to ensure the caller doesn't actively set it to <= 0.
 func WithBlocksize(blocksize int) WriterOpt {
 	return func(w *WriterOpts) error {
-		w.Blocksize = blocksize
+		if blocksize > 0 {
+			w.Blocksize = blocksize
```

Comment:
Should we check for inverse and just throw an error here?

Reply:
I don't understand.

Reply:
Do you mean, rather than ignoring/fixing it if the `blocksize <= 0`, treat `blocksize <= 0` as an error?

Reply:
> treat blocksize <= 0 as an error?

Yes

Reply:
Done.